### PR TITLE
[IL- 308 ] - change password2

### PR DESCRIPTION
**Pull request by TetianaPerinha**
Task's name: [IL- 308 ] Create auto-tests for functional "change password" through a reset
Developer's full name: Tetiana Perinha
Changes:
Add negative tests to the test_auth_change_password.py:
- test_change_password_incorrect_password
- test_reset_password_incorrect_email
- test_reset_password_incorrect_code
Add method for inserting user in DB with randomly generated Gmail - 
- generate_and_insert_user_with_custom_gmail

Refactor:
Change the creation user method  for test_change_password_through_reset_password

### CODE REVIEW

File: tests/authentication/test_auth_change_password.py [hunk 1]
```diff
@@ -28,41 +38,411 @@ class TestResetPassword:
...
-                "imap_server": imap_server2,
-                "email_address_to_connect": email_address_to_connect2,
-                "gmail_password": gmail_password2,
-            }
+            pytest.param(
+                None,
+                None,
+                generate_string(2),
+                400,
+                "Password must be at least 8 characters long and contain at least one letter, one digit, and may include special characters @$!%*?&",
```
Comment (annstriganova) [inline tests/authentication/test_auth_change_password.py]: This message is repeated, i would extract it to a constant, e.g. `INCORRECT_PASSWORD_MSG=...`

File: tests/authentication/test_auth_change_password.py [hunk 2]
```diff
@@ -28,41 +38,411 @@ class TestResetPassword:
...
+    )
+    @pytest.mark.parametrize(
+        "code_for_reset_password,email, new_password, expected_status_code, expected_message_part",
+        [
+            pytest.param(
+                " ",
+                "example.domain.com",
+                generate_password(8),
+                400,
+                "Email must be valid",
```
Comment (annstriganova) [inline tests/authentication/test_auth_change_password.py]: This message is repeated, i would extract it to a constant, e.g. `INCORRECT_EMAIL_MSG=...`

File: tests/authentication/test_auth_change_password.py [hunk 3]
```diff
@@ -28,41 +38,411 @@ class TestResetPassword:
...
+            ),
+            pytest.param(
+                "123-123-123",
```
Comment (annstriganova) [inline tests/authentication/test_auth_change_password.py]: This code is repeated, I would extract it to a constant

**Review by annstriganova — CHANGES_REQUESTED** [review]
I have a couple of suggestions and one question: is it expected that tests are marked as ignored?

image

**Comment by TetianaPerinha** [comment]
> I have a couple of suggestions and one question: is it expected that tests are marked as ignored?
> 
> image

It was marked as ignored because of the bug.                "Bug: https://trello.com/c/ilaf1Luy/19-reset-password-the-app-allows-saving-passwords-that-do-not-meet-the-requirements"

File: tests/authentication/test_auth_change_password.py (hunk 1 above)
Comment (TetianaPerinha) [reply tests/authentication/test_auth_change_password.py]: fixed

File: tests/authentication/test_auth_change_password.py (hunk 2 above)
Comment (TetianaPerinha) [reply tests/authentication/test_auth_change_password.py]: fixed

File: tests/authentication/test_auth_change_password.py (hunk 3 above)
Comment (TetianaPerinha) [reply tests/authentication/test_auth_change_password.py]: fixed

**Review by annstriganova — APPROVED** [review]
LGTM, thank you 👍

**Review by ilsinyakov — COMMENTED** [review]
Approved

**Comment by TetianaPerinha** [comment]
Merge

**Pull request merged**